metrics/metrics.go

```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import "time"
+
+// SinceInMicroseconds gets the time since the specified start in microseconds.
+func SinceInMicroseconds(start time.Time) float64 {
```

Comment:
This is hardly useful too. I see how it has more values than `SinceInSeconds`, but this is mostly hiding the actual type (`time.Duration`) which people should be using rather than floats. I don't think we should suggest people to use this method.

Reply:
`SinceInSeconds` and `SinceInMicroseconds` are used to generate prometheus histogram/summary metrics.
We can find these two util functions have been defined and used in lots of projects. (https://cs.k8s.io/?q=sinceInSeconds&i=nope&files=&repos=)

Reply:
Since Prometheus is the recommended way for k8s metrics, both existing and new projects require these two features.
Rather than defining these two functions everywhere, I think defining them in `k/utils` and let all the projects call them from here is a better implementation.

Reply:
I'm still quite unconvinced, maybe someone would have a different opinion, @thockin ?